### PR TITLE
Extend thread safety documentation

### LINKED ISSUE #78: About Thread Safety
What exactly is the goal of the thread-safe operations? I get it for `basicFlipBit` and `basicUnsafeWrite` (they use a single atomic operation to modify the array), but not `modifyByteArray` (it uses two atomic operations, so it can still happen that the array is modified inbetween). Is it undefined behaviour to have multiple concurrent array writes to the same index?

And why are there `Data.Bit.ImmutableTS` and `Data.Bit.F2PolyTS` modules? These only use local mutability, so it shouldn't make a difference wether they use atomics or not (except that the "thread safe" version is slower).

### PR DESCRIPTION
**Pull request by konsumlamm**
Closes #78.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,12 +16,17 @@ On the other hand, for certain bulk bit operations
 
 ## Thread safety
 
-* `Data.Bit` is faster, but writes and flips are thread-unsafe.
+* `Data.Bit` is faster, but writes and flips are not thread-safe.
   This is because naive updates are not atomic:
   they read the whole word from memory,
   then modify a bit, then write the whole word back.
+  Concurrently modifying separate slices of the same underlying array
```
Comment (Bodigrim) [inline README.md:23]: Maybe “adjacent” instead of “separate”?

Comment (konsumlamm) [reply README.md:23]: "Adjacent" implies that they're directly next to each other, which doesn't have to be the case. I first used "disjoint" (which would be the precise term), but figured that might be too jargony. I realize that "separate" can be misunderstood. Perhaps "non-intersecting"?

Comment (Bodigrim) [reply README.md:23]: “Non-intersecting” is good.

**Comment by Bodigrim** [comment]
Thanks!

**Pull request merged**